### PR TITLE
Create eslint pre-commit hook

### PR DESCRIPTION
**Pull request by sean-usds**
This PR ...
- Creates a shell script pre-commit hook that will run if there are any changes to .tsx or .js files to look for lint errors

Test Steps:
1. Make a format change in a .tsx file that breaks lint convensions
2. Commit the change
3. See that there is a warning 
4. Repeat for .js files

### Testing
- [x] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [x] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Fixes
- #issue - List GitHub issues this PR fixes

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: .environment/eslint/run-eslintCheck.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+function note() {
+    echo "eslint-check> info: ${*}"
+}
+
+function js_fmt_check() {
+    note "Checking JavaScript formatting."
+    MODIFIED_TSX_FILES_COUNT=$(git status --porcelain | grep "\.tsx$" | wc -l)
+    MODIFIED_JS_FILES_COUNT=$(git status --porcelain | grep "\.js$" | wc -l)
```
Comment (TomNUSDS) [inline .environment/eslint/run-eslintCheck.sh:10]: Do we want to limit to the `/src` directory explicitly for these greps? 

I'm wondering if we might hit future issues with some files in `/cypress` or `/public`? Right now it's fine, just trying to think to potential future gotchas. 

It's fine to not change this, just thought I'd mention it as a possibility.

**Review by TomNUSDS — APPROVED** [review]
LGTM; Added a thought to the file matching grep.

File: .environment/eslint/run-eslintCheck.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    if [[ ${MODIFIED_TSX_FILES_COUNT?} != 0 ]] || [[ ${MODIFIED_JS_FILES_COUNT?} != 0 ]]; then
```
Comment (cwinters-usds) [inline .environment/eslint/run-eslintCheck.sh:11]: can you remove the double-brackets here, or change the comparison to `-ne 0`? (see https://github.com/CDCgov/prime-reportstream/pull/3064 for where we had to fix this elsewhere)

File: .environment/eslint/run-eslintCheck.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        cd frontend-react
+        npm run lint
```
Comment (cwinters-usds) [inline .environment/eslint/run-eslintCheck.sh:13]: It looks like you need to have run `yarn install` to run `npm run lint` -- but of course if you're modifying JS code then you should already have run that. So I guess it's ok :-)

**Review by cwinters-usds — APPROVED** [review]
This looks ok to me! :+1:

File: .environment/eslint/run-eslintCheck.sh (hunk 3 above)
Comment (sean-usds) [reply .environment/eslint/run-eslintCheck.sh:13]: couldn't hurt!

**Pull request merged**